src/main/java/jenkins/plugins/git/GitUsernamePasswordBinding.java

```diff
@@ -0,0 +1,182 @@
+package jenkins.plugins.git;
+
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.Extension;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.credentialsbinding.BindingDescriptor;
+import org.jenkinsci.plugins.credentialsbinding.MultiBinding;
+import org.jenkinsci.plugins.credentialsbinding.impl.AbstractOnDiskBinding;
+import org.jenkinsci.plugins.credentialsbinding.impl.UnbindableDir;
+import org.jenkinsci.plugins.gitclient.CliGitAPIImpl;
+import org.jenkinsci.plugins.gitclient.Git;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+public class GitUsernamePasswordBinding extends MultiBinding<StandardUsernamePasswordCredentials> implements GitCredentialBindings {
+    final static private String GIT_USERNAME_KEY = "GIT_USERNAME";
+    final static private String GIT_PASSWORD_KEY = "GIT_PASSWORD";
+    final private String gitToolName;
+    final private Map<String, String> credMap = new LinkedHashMap<>();
+    private String gitToolExe = null;
+    private boolean unixNodeType;
```

Comment:
I think that this field should be declared `transient` so that it is not stored in the job definition.  I believe the storage in the field is used to pass a value into setRunEnvironmentVariables but should not be considered persistent across jobs.  It is computed based on the launcher used for the current job and the computed value is used immediately to prepare the environment for the credential binding.
```suggestion
    private transient boolean unixNodeType;
```